### ISSUE TITLE
 Inquiry about aspect_loss in training

### ISSUE THREAD

**Issue by WuYunfan** [hongyuntw/RBRS-ARG#1]
Hi,

I have been reading the code you provided and have a question regarding the implementation of `aspect_loss` in the training process.

From my understanding, the `sample_reviews_aspect_emb` is calculated using `review_ids`, which are integer indices. Since these indices are used in a lookup table operation, it seems that gradients cannot propagate back to the generation model, making `aspect_loss` ineffective in updating the model parameters.

Here is the relevant part of the code in train_arg.py:

```python
sample_reviews_aspect_emb = get_aspect_emb(sample_output_texts, opt, w2v, abae, mode='sample')
input_review_aspect_emb = get_aspect_emb(input_reviews, opt, w2v, abae, mode='input')
aspect_loss = l1_loss(sample_reviews_aspect_emb, target_aspect_embedding)
loss = rl_loss + aspect_loss
```

Given that `review_ids` in `get_aspect_emb`  are integers, it appears that the gradients cannot flow back to the generator model through `aspect_loss`.

Could you please confirm if my understanding is correct? If so, do you have any suggestions on how to modify the code to ensure that `aspect_loss` can effectively update the generator model parameters?

Thank you for your time and assistance. I look forward to your guidance.

Best regards,
